pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java

```diff
@@ -1660,6 +1661,26 @@ public static MessageMetadata peekMessageMetadata(ByteBuf metadataAndPayload, St
         }
     }
 
+    public static final String NONE_KEY = "NONE_KEY";
+    public static byte[] peekStickyKey(ByteBuf metadataAndPayload, String topic, String subscription) {
+        try {
+            int readerIdx = metadataAndPayload.readerIndex();
+            skipBrokerEntryMetadataIfExist(metadataAndPayload);
+            MessageMetadata metadata = Commands.parseMessageMetadata(metadataAndPayload);
+            metadataAndPayload.readerIndex(readerIdx);
+            byte[] key = NONE_KEY.getBytes();
```

Comment:
I think we should also use `getBytes(StandardCharsets.UTF_8)` here.

Reply:
If there is no key, instead of calling `getBytes()` each time, we should just keep a static ref to the `byte[]`